src/apps/replication/lib/replica_learn.cpp

```diff
@@ -242,10 +244,11 @@ void replica::on_learn(dsn_message_t msg, const learn_request& request)
     }
 
     // mutations are previously committed already on learner (old primary)
+    // TODO(qinzuoyan): this case will never occur?
```

Comment:
when the old primary comes back, this is possible. The state is always consistent in this case as we always replay the prepare list.